test/loki_service.cc

```diff
@@ -22,6 +23,59 @@ using namespace prime_server;
 
 namespace {
 
+boost::property_tree::ptree make_config(const std::vector<std::string>& whitelist = {
+                                            "locate",
+                                            "route",
+                                            "height",
+                                            "sources_to_targets",
+                                            "optimized_route",
+                                            "isochrone",
+                                            "trace_route",
+                                            "trace_attributes",
+                                            "transit_available",
+                                            "expansion",
+                                            "centroid",
+                                            "status",
+                                        }) {
+  auto run_dir = VALHALLA_BUILD_DIR "test" + std::string(1, filesystem::path::preferred_separator) +
+                 "loki_service_tmp";
+  if (!filesystem::is_directory(run_dir) && !filesystem::create_directories(run_dir))
+    throw std::runtime_error("Couldnt make directory to run from");
+
+  auto config = test::make_config(run_dir,
+                                  {
+                                      {"service_limits.skadi.max_shape", "100"},
+                                      {"service_limits.max_exclude_locations", "0"},
+                                  },
+                                  {"loki.actions"});
+
+  boost::property_tree::ptree actions;
+  for (const auto& action_name : whitelist) {
+    boost::property_tree::ptree action;
+    action.put("", action_name);
+    actions.push_back(std::make_pair("", action));
+  }
+  config.add_child("loki.actions", actions);
+
+  return config;
+}
+
+std::string write_config(boost::property_tree::ptree pt) {
+  // config for permanently running server and the status response
+  std::stringstream config_stream;
+  for (const auto& path : {"mjolnir.tile_dir", "mjolnir.tile_extract", "mjolnir.admin",
+                           "mjolnir.timezone", "mjolnir.transit_dir", "additional_data.elevation"}) {
+    pt.erase(path);
+  }
+  boost::property_tree::json_parser::write_json(config_stream, pt);
+
+  return config_stream.str();
+}
+
+// config for permanently running server
+auto const config = make_config();
+auto const config_str = write_config(config);
```

Comment:
just moved that whole block up from below (and added `config_str`) since I need the config as string in the response definition for `/status`